### PR TITLE
Caff 80

### PR DESCRIPTION
**Pull request by 1999michael**
holdup, lemme look at the code and make it PR-able

### CODE REVIEW

File: description/launch/simulate.launch [hunk 1]
```diff
@@ -5,9 +5,9 @@
     <arg name="rviz" default="false"/>
 
     <!-- IGVC worlds: full, walls, ramp, plain -->
-    <!-- IGVC world types: pavement (2022 IGVC), grass (<2021 IGVC)-->
+    <!-- IGVC world types: pavement (2022 IGVC), grass (<2021 IGVC), plain (pavement testing)-->
     <arg name="world" default="full"/>
-    <arg name="world_type" default="grass"/>
+    <arg name="world_type" default="pavement"/>
```
Comment (kajananchinniah) [inline description/launch/simulate.launch:10]: Could you pull master? This will likely be a merge conflict

File: description/urdf/constants.xacro [hunk 2]
```diff
@@ -82,9 +82,9 @@
     <xacro:property name="laser_horizontal_min_angle" value="${-0.5*laser_field_of_view*pi/180}"/>
     <xacro:property name="laser_horizontal_max_angle" value="${0.5*laser_field_of_view*pi/180}"/>
 
-    <xacro:property name="laser_range_min" value="0.06"/>
+    <xacro:property name="laser_range_min" value="0.1"/>
```
Comment (kajananchinniah) [inline description/urdf/constants.xacro:85]: Why did you change the laster min ?

File: description/urdf/constants.xacro [hunk 3]
```diff
@@ -108,7 +108,7 @@
     <xacro:property name="ZED_camera_length" value="0.175"/>
     <xacro:property name="ZED_camera_width" value="0.03"/>
     <xacro:property name="ZED_camera_height" value="0.033"/>
-    <xacro:property name="ZED_camera_pitch" value="${15 * pi/180}"/>
+    <xacro:property name="ZED_camera_pitch" value="${19.66 * pi/180}"/> <!-- 19.66 degrees ensures the camera's field of view overlaps with the end of the lidar readings-->
```
Comment (kajananchinniah) [inline description/urdf/constants.xacro:111]: Is 19.96 deg necessary? We should inform mech asap if yes

File: nav/load_waypoints/scripts/navigate_waypoints.py [hunk 4]
```diff
@@ -100,7 +106,7 @@ def wait_for_utm_transform(self):
     
     def get_next_waypoint(self):
         waypoint = self.waypoints[self.curr_waypoint_idx]
-        print self.curr_waypoint_idx
+        print(self.curr_waypoint_idx)
         self.curr_waypoint_idx += 1
```
Comment (kajananchinniah) [inline nav/load_waypoints/scripts/navigate_waypoints.py:110]: Could you make the waypoints wrap around? We might need this for the second lap

File: nav/load_waypoints/scripts/navigate_waypoints.py [hunk 5]
```diff
@@ -56,8 +59,11 @@ def populate_waypoint_dict(self):
             gps_info = rospy.wait_for_message('gps/fix', NavSatFix)
             # Append the starting gps coordinate to the waypoints dict as the final waypoint
             last_coord_idx = len(self.waypoints) 
-            self.waypoints[last_coord_idx] = {
-                'id': last_coord_idx, 'longitude': gps_info.longitude, 'latitude': gps_info.latitude, 'description': 'Initial start location', 'frame_id': 'odom'}
+ 
+            # If waypoints are reported in longitudes and latitudes, append a final waypoint to return to the start
+            if self.waypoints[0].get("longitude"):
```
Comment (kajananchinniah) [inline nav/load_waypoints/scripts/navigate_waypoints.py:64]: Wyhy was this change necessary ?

File: nav/load_waypoints/scripts/navigate_waypoints.py [hunk 6]
```diff
@@ -121,8 +127,18 @@ def get_pose_from_gps(self, longitude, latitude, frame, pose_test_var = None):
...
-        
-        
+    def get_pose_from_xy(self, x, y):
+
+        # Convert (x,y) coordinate to pose
+        # (x,y) coordinate follows the coordinate system used in spawn.launch
+
+        xy_pose = PoseStamped()
+        xy_pose.header.frame_id = 'odom' # not sure what the proper frame is
+        xy_pose.pose.position.x = y - 0 # subtract starting y-coordinate
```
Comment (kajananchinniah) [inline nav/load_waypoints/scripts/navigate_waypoints.py:137]: What are these hardcoded params? Could you make a note (we need to know if we need to adjust them for the actual comp)

File: nav/load_waypoints/scripts/navigate_waypoints.py [hunk 7]
```diff
@@ -148,7 +167,7 @@ def send_and_wait_goal_to_move_base(self, curr_waypoint):
         action_client.send_goal(goal)
 
         # Waits for the server to finish performing the action.
-        finished_within_time = action_client.wait_for_result(rospy.Duration(60))
+        finished_within_time = action_client.wait_for_result(rospy.Duration(6000))
```
Comment (kajananchinniah) [inline nav/load_waypoints/scripts/navigate_waypoints.py:170]: Just to double check, this returns early if the result is done right? The Duration is just a timeout

File: nav/nav_stack/CMakeLists.txt [hunk 8]
```diff
@@ -9,3 +9,5 @@ find_package(catkin REQUIRED COMPONENTS
   std_msgs
   tf
 )
+
+add_definitions(-DROSCPP_USE_TCP_NODELAY)
```
Comment (kajananchinniah) [inline nav/nav_stack/CMakeLists.txt:13]: What's this for?

File: nav/load_waypoints/scripts/static_waypoints.json [hunk 9]
```diff
@@ -1,44 +1,14 @@
-{ 
+{
```
Comment (kajananchinniah) [inline nav/load_waypoints/scripts/static_waypoints.json:1]: Why'd you delete most of this file ? We need to have all 4 waypoints of no man's land

File: nav/load_waypoints/scripts/static_waypoints_xy.json [hunk 10]
```diff
@@ -0,0 +1,14 @@
+{
```
Comment (kajananchinniah) [inline nav/load_waypoints/scripts/static_waypoints_xy.json:1]: How come we have two static waypoint jsons now?

**Review by kajananchinniah — CHANGES_REQUESTED** [review]
I can't view changes done to the worlds. But why was the world changed?

Other than that, lgtm I guess, due to time I won't be too strict on review

File: nav/load_waypoints/scripts/navigate_waypoints.py (hunk 5 above)
Comment (1999michael) [reply nav/load_waypoints/scripts/navigate_waypoints.py:64]: if you mean the if statement in line 64, its the wraparound we need

File: nav/load_waypoints/scripts/navigate_waypoints.py (hunk 4 above)
Comment (1999michael) [reply nav/load_waypoints/scripts/navigate_waypoints.py:115]: Essentially no change other than style now

File: nav/load_waypoints/scripts/navigate_waypoints.py (hunk 7 above)
Comment (1999michael) [reply nav/load_waypoints/scripts/navigate_waypoints.py:153]: yes duration is just timeout (in seconds)

File: nav/load_waypoints/scripts/navigate_waypoints.py (hunk 4 above)
Comment (kajananchinniah) [reply nav/load_waypoints/scripts/navigate_waypoints.py:115]: Wrap around to prevent crashing. On the final index, we'll get index out of bounds

Also, is the print change correct? Python2 expects the old change

Comment (1999michael) [reply nav/load_waypoints/scripts/navigate_waypoints.py:115]: I think I did it cause I had backend python3 default for some time. I don't think ( ) will make a diff even in python2

**Review by kajananchinniah — APPROVED** [review]
Approved but please deal with the comments I left

File: nav/nav_stack/CMakeLists.txt (hunk 8 above)
Comment (1999michael) [reply nav/nav_stack/CMakeLists.txt]: removed

File: nav/load_waypoints/scripts/navigate_waypoints.py (hunk 6 above)
Comment (1999michael) [reply nav/load_waypoints/scripts/navigate_waypoints.py]: removed the xy stuff

File: description/urdf/constants.xacro (hunk 2 above)
Comment (1999michael) [reply description/urdf/constants.xacro]: just testing stuff. reverted back
